guppylang/definition/custom.py

```diff
@@ -97,10 +97,16 @@ def compile_call(
 
 
 @dataclass(frozen=True)
-class CustomFunctionDef(RawCustomFunctionDef, CompiledCallableDef):
+class CustomFunctionDef(CompiledCallableDef):
     """A custom function with parsed and checked signature."""
 
+    defined_at: AstNode
+    call_checker: "CustomCallChecker"
+    call_compiler: "CustomCallCompiler"
     ty: FunctionType
+    higher_order_value: bool
```

Comment:
When is this not true?

Reply:
When the type of the function cannot be expressed in Guppy's type system.

For example, the `len(x)` function is a `CustomFunction` that checks if `x` implements `__len__`. We can't write down a type for `len`, so we can't use it as a value